src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -227,6 +227,11 @@ protected virtual bool TryGetKeyIdentifierHeader(out string keyIdentifier, IInco
             return context.Headers.TryGetValue(Headers.RijndaelKeyIdentifier, out keyIdentifier);
         }
 
+        protected virtual void GenerateIV(RijndaelManaged rijndael)
```

Comment:
I agree it can be abused, but at the moment this risk is limited to unit test suite.

First approach will yield more work because we'll need to set up `RijndaelManager` for encryption and decryption separately and will introduce a chance for our users to make a mistake.

Second approach precludes the simplest option, which is to call `RijndaelManaged.GenerateIV` method, which is what would happen in majority of cases.